### PR TITLE
VYD-2957: 191212 rescue validation error and append to errors

### PR DESCRIPTION
**Pull request by gzssince1990**
[VYD-2957]

[VYD-2957]: https://vydia.atlassian.net/browse/VYD-2957

### CODE REVIEW

File: lib/gourami/extensions/resources.rb [hunk 1]
```diff
@@ -91,6 +91,28 @@ def any_resource_errors?
...
+        new_resource_errors = new_resource_errors.dup
+        resource_errors.clear
+        resource_errors.merge!(new_resource_errors)
+
+        resource_errors
+      end
+
+      def handle_validation_error(error)
+        super(error)
+        clear_and_set_resource_errors(error.resource_errors) unless error.resource_errors.nil?
```
Comment (TSMMark) [inline lib/gourami/extensions/resources.rb:109]: `unless error.resource_errors.nil?` What's that for?

File: lib/gourami/validations.rb [hunk 2]
```diff
@@ -57,6 +62,14 @@ def clear_and_set_errors(new_errors)
       errors
     end
 
+    def raise_validate_errors
+      raise ValidationError.new(errors)
+    end
+
+    def handle_validation_error(error)
+      clear_and_set_errors(error.errors) unless error.errors.nil?
```
Comment (TSMMark) [inline lib/gourami/validations.rb:70]: I don't think this `unless error.errors.nil?` stuff is right.

File: lib/gourami/extensions/resources.rb (hunk 1 above)
Comment (gzssince1990) [reply lib/gourami/extensions/resources.rb:109]: This method could be called without any resource errors right?

File: lib/gourami/extensions/resources.rb [hunk 3]
```diff
@@ -91,6 +91,28 @@ def any_resource_errors?
...
+      end
+
+      def raise_validate_errors
+        raise ValidationError.new(errors, resource_errors)
```
Comment (TSMMark) [inline lib/gourami/extensions/resources.rb:113]: Awesome

File: lib/gourami/validations.rb (hunk 2 above)
Comment (gzssince1990) [reply lib/gourami/validations.rb:70]: Are you saying that we don't need it?

**Review by TSMMark — APPROVED** [review]
Nice

File: lib/gourami/validations.rb (hunk 2 above)
Comment (TSMMark) [reply lib/gourami/validations.rb:70]: I read it wrong, I was thinking it said `self.errors.nil?`

**Pull request merged**